### ISSUE TITLE
DESCRIPTION should require minimum version for `raster` to avoid test failures

### ISSUE THREAD

**Issue by barracuda156** [rstudio/leaflet#881]
```
R version 4.3.0 (2023-04-21) -- "Already Tomorrow"
Copyright (C) 2023 The R Foundation for Statistical Computing
Platform: powerpc-apple-darwin10.8.0 (32-bit)

> library(testthat)
> library(leaflet)
> 
> test_check("leaflet")
The legacy packages maptools, rgdal, and rgeos, underpinning this package
will retire shortly. Please refer to R-spatial evolution reports on
https://r-spatial.org/r/2023/05/15/evolution4.html for details.
This package is now running under evolution status 0 
terra 1.7.46

Attaching package: 'terra'

The following objects are masked from 'package:testthat':

    compare, describe

Loading required package: sp
[ FAIL 1 | WARN 0 | SKIP 7 | PASS 136 ]

══ Skipped tests ═══════════════════════════════════════════════════════════════
• On CRAN (2)
• sf cannot be loaded (5)

══ Failed tests ════════════════════════════════════════════════════════════════
── Error ('test-raster.R:25:3'): rasters ───────────────────────────────────────
Error in `.rawTransform(projfrom, projto, xy)`: no slot of name "ptr" for this object of class "SpatVector"
Backtrace:
    ▆
 1. └─leaflet::projectRasterForLeaflet(raster(lux), "bilinear") at test-raster.R:25:2
 2.   ├─raster::projectRaster(...)
 3.   │ └─raster:::.getSRS(to)
 4.   │   └─methods::extends(class(x), "CRS")
 5.   └─raster::projectExtent(x, crs = sp::CRS(epsg3857))
 6.     └─raster:::.rawTransform(projfrom, projto, xy)

[ FAIL 1 | WARN 0 | SKIP 7 | PASS 136 ]
Error: Test failures
Execution halted
```

P. S. `sf` tests skipped, since `sf` does not build in Rosetta, and I am away from the native hardware atm.

**Comment by gadenbuie** [comment]
Can you provide the complete results of `devtools::session_info()`? It would be helpful to know which versions of `sp`, `raster` and others you're using here. Thanks!

**Comment by barracuda156** [comment]
Let me update relevant packages and re-run tests. Maybe it is just that I have outdated versions on this specific machine.

**Comment by barracuda156** [comment]
@gadenbuie Indeed, the issue was with some outdated dependency (`raster`, perhaps).
After updates, all tests pass.

However, perhaps `DESCRIPTION` should be updated to require a minimum version for it?

**Comment by barracuda156** [comment]
@gadenbuie Another issue, apparently tests do not get installed for some reason. I.e. they run fine from command line via Macports, but when I try to run them from R session, they are not found:
```
> library(testthat)
> library(leaflet)
> test_check("leaflet")
Error in `test_dir()`:
! No test files found
Run `rlang::last_trace()` to see where the error occurred.

> rlang::last_trace(drop = FALSE)
<error/rlang_error>
Error in `test_dir()`:
! No test files found
---
Backtrace:
    ▆
 1. └─testthat::test_check("leaflet")
 2.   └─testthat::test_dir("testthat", package = package, reporter = reporter, ..., load_package = "installed")
 3.     └─rlang::abort("No test files found")
```

**Comment by gadenbuie** [comment]
> @gadenbuie Indeed, the issue was with some outdated dependency (`raster`, perhaps).
> After updates, all tests pass.

Do you know which packages were updated in this? One guess is that it might be `sp` that caused the issue since you updated from `1.6-1` and the current version on CRAN is `2.0-0`.

But the code in the failing test was written about 14 months ago and the initial versions of raster and sp that you have were both released much more recently (raster 3.6-20 was released 2023-03 and sp 1.6-1 was released 2023-05-31). So this is feeling like it's much more likely that another package beside leaflet should be setting the minimum version requirement.

**Comment by barracuda156** [comment]
@gadenbuie In fact I have overlooked update for `sp`, so it is definitely not because of it. I can revert updates selectively and re-run tests; anything besides `raster` to suspect? Blind search will be rather tedious.
